lib/types/p2p.ts

```diff
@@ -9,6 +9,7 @@ export type Address = {
 export type NodeConnectionInfo = {
   nodePubKey: string;
   addresses: Address[];
+  lastAddress?: Address | undefined;
```

Comment:
Is the `| undefined` necessary here, after you made `lastAddress` optional?

Reply:
I guess you're right, probably I overlooked it